test/common/http/http1/codec_impl_test.cc

```diff
@@ -1352,24 +1352,12 @@ TEST_F(Http1ClientConnectionImplTest, FlowControlReadDisabledReenable) {
   std::string output;
   ON_CALL(connection_, write(_, _)).WillByDefault(AddBufferToString(&output));
 
-  // 1st pipeline request.
+  // Request.
   TestRequestHeaderMapImpl headers{{":method", "GET"}, {":path", "/"}, {":authority", "host"}};
   request_encoder->encodeHeaders(headers, true);
   EXPECT_EQ("GET / HTTP/1.1\r\nhost: host\r\ncontent-length: 0\r\n\r\n", output);
   output.clear();
 
-  // 2nd pipeline request.
```

Comment:
For my own knowledge, why was this removed in this PR?

Reply:
When the header maps are fully split from an interface perspective, the existing code did not work very well as we had the wrong header type at various places in the base class. This was especially problematic with HEAD requests. I could have probably fixed this a different way, but I realized that if I completely removed pipelining support, fixing became a lot easier. This also removes all heap allocations for HTTP/1 requests and responses which is a pretty nice perf benefit. Given that we don't actually use pipelining anywhere in the real code, this seemed like a good simplification to make.

BTW this is why I split this change out into its own PR because it definitely needs a good review.